sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java

```diff
@@ -156,7 +158,7 @@ public KeyVaultAccessControlAsyncClient buildAsyncClient() {
         // Add retry policy.
         policies.add(retryPolicy == null ? new RetryPolicy() : retryPolicy);
 
-        policies.add(new KeyVaultCredentialPolicy(credential));
+        policies.add(new BearerTokenAuthenticationPolicy(credential, MANAGED_HSM_SCOPE));
```

Comment:
In order to utilize the Challenge Based Auth Support in BTA policy from azure-core
we need to create a custom policy in KV package that doesn't hard code the scopes as above.
The scopes and tenant id are dynamically parsed in KV challenge that gets returned back from the service.
Here's the .NET impl for reference: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs#L13

You can port that over to Java and then utilize that policy here, instead of above.

cc: @schaabs @AlexGhiondea @heaths

Reply:
Thanks @g2vinay!